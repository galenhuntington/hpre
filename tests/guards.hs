{-# LANGUAGE LambdaCase #-}

isPos :: Int -> Bool
''  x | x > 0 = True
      |       = False

isPos' :: Int -> Bool
''  = \case
   x | x > 0 -> True
     |       -> False
