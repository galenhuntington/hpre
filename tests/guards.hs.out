{-# LANGUAGE LambdaCase #-}

isPos :: Int -> Bool
isPos x | x > 0 = True
        | True  = a where a :: Bool
                          a = False

isPos' :: Int -> Bool
isPos' = \case
   x | x>0 -> True
     | True-> a where a :: Bool
                      a = False

noQuote :: String
noQuote =
   reverse "|  ="
   ++
   reverse "|  ->"

